### PR TITLE
Fix scale_factor/add_offset with short/int var

### PR DESCRIPTION
**Pull request by benjwadams**
Fixes improper comparison short/int variable against floating point
scale_factor/add_offset.

### CODE REVIEW

**Comment by mhidas** [comment]
Thanks @benjwadams - that fixes the issue for the example file I provided in https://github.com/ioos/compliance-checker/issues/842#issuecomment-1025334905
Unfortunately it seems one example was not enough. I still get the incorrect error for [this file](https://s3-ap-southeast-2.amazonaws.com/imos-data/IMOS/SRS/Surface-Waves/SAR/DELAYED/SENTINEL-1A/2020/01/IMOS_SRS-Surface-Waves_W_20200101T000011Z_Sentinel-1A_FV01_DM00_K2_END-20200101T001209Z.nc), e.g. for this variable:
```
        short HEADING(TIME) ;
                HEADING:add_offset = 180.f ;
                HEADING:scale_factor = 0.01f ;
```
The difference from the other file is the attributes are now floats instead of double, but looking at the updated code I can't see why that fails.

**Review by greenteadigital — APPROVED** [review]

**Comment by benjwadams** [comment]
@mhidas, this should fix.  `float` in numpy isn't the same as C `float` (usually 32 bits), so the data needed an explicit comparison against NumPy's `float32`.

**Review by mhidas — APPROVED** [review]
:+1: This fixes all the issues I had with incorrect checker results for my files.

The other thing I noticed in the tests is a separate issue that would reuquire extending the current checks - up to you if you want to address it.

File: compliance_checker/tests/test_cf.py [hunk 1]
```diff
@@ -2261,8 +2261,8 @@ def test_check_add_offset_scale_factor_type(self):
             coarse_temp = dataset.createVariable(f"coarse_temp_{var_bytes}",
                                                  f"i{var_bytes}",
                                                  dimensions=("time",))
-            coarse_temp.setncattr("scale_factor", np.float(23.0))
-            coarse_temp.setncattr("add_offset", np.float(-2.1))
+            coarse_temp.setncattr("scale_factor", np.float32(23.0))
+            coarse_temp.setncattr("add_offset", np.double(-2.1))
```
[the remark this answers is not in the document]
Comment (benjwadams) [reply compliance_checker/tests/test_cf.py:2265]: I'll fix this as well.  The current check isn't stateful in this regard.

[the remark this answers is not in the document]
Comment (benjwadams) [reply compliance_checker/tests/test_cf.py:2265]: Addressed in 4b46 below.  Doesn't explicitly check float or double by itself, but if variables are the wrong type, that will be picked up by the previous checks anyhow.

[the remark this answers is not in the document]
Comment (mhidas) [reply compliance_checker/tests/test_cf.py:2265]: :+1: Looks good to me! Thanks!

**Review by mhidas — APPROVED** [review]

**Pull request merged**